src/Model/Behavior/UploadBehavior.php

```diff
@@ -187,12 +192,12 @@ public function getPathProcessor(EntityInterface $entity, $data, string $field,
      * Retrieves an instance of a file writer which knows how to write files to disk
      *
      * @param \Cake\Datasource\EntityInterface $entity an entity
-     * @param array $data the data being submitted for a save
+     * @param \Psr\Http\Message\UploadedFileInterface|array $data the data being submitted for a save
      * @param string $field the field for which data will be saved
      * @param array $settings the settings for the current field
      * @return \Josegonzalez\Upload\File\Writer\WriterInterface
      */
-    public function getWriter(EntityInterface $entity, array $data, string $field, array $settings): WriterInterface
+    public function getWriter(EntityInterface $entity, $data, string $field, array $settings): WriterInterface
```

Comment:
Could this be type hinted for the interface?

Reply:
The problem is call ``getWriter`` on ``afterDelete`` we dont have an UploadedFileInterface concrete object there like on ``beforeSave``.  Should i create an empty UploadedFile object?

Reply:
Well you've added a type hint to the `WriterInterface` so passing array into this method would cause a type error. So I'd suggest either using `WriterInterface|array` in the docblock, or sticking just with `WriterInterface`.

My personal preference is always `null` rather than an empty object to allow for `?WriterInterface` type hint, but see what the other reviewers think. I haven't had a huge involvement in this project 😄

Reply:
I honestly don't mind either way, so lets do what @davidyell said since that seems reasonable.